azkaban-common/src/test/java/azkaban/executor/ContainerizedDispatchImplTest.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import azkaban.Constants;
+import azkaban.Constants.ConfigurationKeys;
+import azkaban.Constants.ContainerizedExecutionManagerProperties;
+import azkaban.executor.container.ContainerizedDispatchImpl;
+import azkaban.executor.container.ContainerizedImpl;
+import azkaban.executor.container.ContainerizedImplType;
+import azkaban.metrics.CommonMetrics;
+import azkaban.metrics.MetricsManager;
+import azkaban.user.User;
+import azkaban.utils.Pair;
+import azkaban.utils.Props;
+import azkaban.utils.TestUtils;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ContainerizedDispatchImplTest {
```

Comment:
A number of tests here seem to be generic enough the can moved to a test class which will be common for both ExecutionController and ContainerizedDispatch, something like `AbstractExecutionManagerAdapterTest` .
This class can instead focus on any tests specific to containerization.
Let me know your thoughts.

Reply:
I agree with you. I have created a ticket for the same. https://jira01.corp.linkedin.com:8443/browse/LIHADOOP-56544